gapis/service/service.proto

```diff
@@ -671,6 +672,76 @@ message MemoryRange {
   uint64 size = 2;
 }
 
+message Metrics {
+  oneof metrics {
+    MemoryLayout memory_layout = 1;
+  }
+}
+
+// The description of the memory layout of the API state
+message MemoryLayout {
+  // The API ID used for this call.
+  path.API api = 1;
+  // The index to get the constants names for MemoryAllocation `flags`.
+  // A value of -1 indicates no flag names should be fetched.
+  int32 allocation_flags_index = 2;
+  // The memory allocations that are currently in use.
+  repeated MemoryAllocation allocations = 3;
+}
+
+// A single memory allocation
+message MemoryAllocation {
+  // The device this allocation was made on
+  uint64 device = 1;
+  // The memory type this allocation is from
+  uint32 memory_type = 2;
+  // Flags describing properties of the allocation, e.g. whether it's device
+  // local, whether it's host visible, etc.
+  uint32 flags = 3;
+  // Constant index for looking up the flag values with the path.ConstantSet endpoint
+  uint32 constant_set = 4;
+  // The API specific ID for this allocation
+  uint64 id = 5;
```

Comment:
Is this Id used? And should we name it 'handle'?

Reply:
@ben-clayton suggested separated id and name for bindings, wasn't sure if it would also be useful for allocations.  Open to using either id or handle, not sure which is a more descriptive name.

Reply:
I have a small preference for Handle. It more closely maps to how I view things. (That is not to say my view is right).

Reply:
Would you prefer to also keep `name` or is just handle good enough?

Reply:
I think keeping the name is fine. It means we can use debug names if assigned.